framework_agreement/model/purchase.py

```diff
@@ -26,47 +26,42 @@ class PurchaseOrder(models.Model):
         'framework.agreement.portfolio',
         'Portfolio',
         domain="[('supplier_id', '=', partner_id)]",
+        help='When a portfolio is selected, the pricelist selection is '
+        'restricted to agreements belonging to it. Moreover, some fields '
+        'of the order cannot be changed because they their values are set in '
+        'the agreement.'
     )
 
     @api.onchange('pricelist_id')
-    def update_currency_from_pricelist(self):
-        """Reproduce the old_api onchange_pricelist from the purchase module.
-
-        We need new-style onchanges to be able to modify agreements on order
-        lines, and we cannot have new-style and old-style onchanges at the same
-        time.
-
-        """
-        self.currency_id = self.pricelist_id.currency_id
-
-    @api.onchange('portfolio_id', 'pricelist_id', 'date_order', 'incoterm_id')
-    def update_agreements_in_lines(self):
-        Agreement = self.env['framework.agreement']
-        if self.portfolio_id:
-            for line in self.order_line:
-                ag_domain = Agreement.get_agreement_domain(
-                    line.product_id.id,
-                    line.product_qty,
-                    self.portfolio_id.id,
-                    self.date_order,
-                    self.incoterm_id.id,
-                )
-                good_agreements = Agreement.search(ag_domain).filtered(
-                    lambda a: a.has_currency(self.currency_id))
-
-                if line.framework_agreement_id in good_agreements:
-                    pass  # it's good! let's keep it!
-                else:
-                    if len(good_agreements) == 1:
-                        line.framework_agreement_id = good_agreements
-                    else:
-                        line.framework_agreement_id = Agreement
-
-                if line.framework_agreement_id:
-                    line.price_unit = line.framework_agreement_id.get_price(
-                        line.product_qty, self.currency_id)
-        else:
-            self.order_line.write({'framework_agreement_id': False})
+    @api.multi
+    def propagate_agreement_fields(self):
+        PROPAGATE_FIELDS = [
+            'currency_id',
+            'payment_term_id',
+            'terms_of_payment',
+            'incoterm_id',
+            'incoterm_address',
+            'origin_address_id',
+            'dest_address_id',
+            'picking_type_id',
+        ]
+        agreement = self.pricelist_id
+
+        for field_name in PROPAGATE_FIELDS:
+            # self.write does not work in an onchange
+            field_value = agreement[field_name]
+            if field_value:
+                self[field_name] = field_value
```

Comment:
I see you added a dependency on purchase_transport_multi_address. This strikes me a something that had better be put in a glue module, instead of including all of this in the basic framework module. My customer is interested in framework agreements without these extra flows. Would you consider such a refactoring?